cmd/erasure-server-pool-decom.go

```diff
@@ -699,27 +703,45 @@ func (v versionsSorter) reverse() {
 }
 
 func (set *erasureObjects) listObjectsToDecommission(ctx context.Context, bi decomBucketInfo, fn func(entry metaCacheEntry)) error {
-	disks := set.getOnlineDisks()
+	disks, _ := set.getOnlineDisksWithHealing(false)
 	if len(disks) == 0 {
 		return fmt.Errorf("no online drives found for set with endpoints %s", set.getEndpoints())
 	}
 
-	listQuorum := (len(disks) + 1) / 2
+	askDisks := getListQuorum("strict", set.setDriveCount)
+	var fallbackDisks []StorageAPI
+
+	// Special case: ask all disks if the drive count is 4
+	if set.setDriveCount == 4 || askDisks > len(disks) {
+		askDisks = len(disks) // use all available drives
+	}
+
+	// However many we ask, versions must exist on ~50%
+	listingQuorum := (askDisks + 1) / 2
```

Comment:
Not sure what you are talking about. This code is exactly like our listObjects() - are you saying that is broken?

Reply:
Listing quorum never be that this is strict quorum.

Reply:
> askDisks == number of drives in a set

Not all the time. This is because of this:

```
	if set.setDriveCount == 4 || askDisks > len(disks) {
		askDisks = len(disks)
	}
```

and 

```
disks, _ := set.getOnlineDisksWithHealing(false)
```

where disks do not include healing disks

Reply:
okay askDisks check is completely removed.